CRM/Campaign/Form/Petition.php

```diff
@@ -126,6 +126,10 @@ public function preProcess() {
    *   array of default values
    */
   public function setDefaultValues() {
+    if ($this->_cdType) {
+      return CRM_Custom_Form_CustomData::setDefaultValues($this);
```

Comment:
Yeah this is a pr against 4.6.  It might well need a different set of changes for 4.7.